pandas/core/arrays/arrow/array.py

```diff
@@ -734,7 +734,14 @@ def _cmp_method(self, other, op) -> ArrowExtensionArray:
             raise NotImplementedError(
                 f"{op.__name__} not implemented for {type(other)}"
             )
-        return ArrowExtensionArray(result)
+        result = ArrowExtensionArray(result)
+        if self.dtype.na_value is np.nan:  # type: ignore[comparison-overlap]
```

Comment:
Semantically wouldn't this be better defined in `ArrowStringArray`? Since `self.dtype` must not be a `ArrowDtype` here

Reply:
either way works; i preferred this since it was maximally high in the inheritance chain

Reply:
Yeah I would prefer this in `ArrowStringArray` since this logic is specific to that subclass.